packages/nbmodel/src/api.ts

```diff
@@ -81,6 +81,10 @@ export interface ISharedBaseCell<Metadata extends nbformat.IBaseCellMetadata>
   extends IDisposable {
   getSource(): string;
   setSource(value: string): void;
+  /**
+   * Replace content from `start' to `end` with `value`.
+   */
+  modifySource(start: number, end: number, value?: string): void;
```

Comment:
maybe updateSource?

Reply:
or changeSource (as we discussed...)?

Reply:
btw other candidate is mutateSource - But it should the same name as the exiting method to update/change/mutate the metadata.